charmhelpers/core/sysctl.py

```diff
@@ -31,18 +31,22 @@
 def create(sysctl_dict, sysctl_file):
     """Creates a sysctl.conf file from a YAML associative array
 
-    :param sysctl_dict: a YAML-formatted string of sysctl options eg "{ 'kernel.max_pid': 1337 }"
+    :param sysctl_dict: a dict or YAML-formatted string of sysctl
+                        options eg "{ 'kernel.max_pid': 1337 }"
     :type sysctl_dict: str
     :param sysctl_file: path to the sysctl file to be saved
     :type sysctl_file: str or unicode
     :returns: None
     """
-    try:
-        sysctl_dict_parsed = yaml.safe_load(sysctl_dict)
-    except yaml.YAMLError:
-        log("Error parsing YAML sysctl_dict: {}".format(sysctl_dict),
-            level=ERROR)
-        return
+    if type(sysctl_dict) is not dict:
```

Comment:
You might consider using `if not isinstance(sysctl_dict, collections.Mapping):` instead, on the off chance that the charm wants to pass in, e.g., an `OrderedDict` instead.  Not really a blocker, though, since the order wouldn't be preserved with the existing code, either.

Reply:
Indeed, `collections.Mapping` is Py2 compatible.

Reply:
importing collections.Mapping raises a warning under Py3.7, and should be collections.abc.Mapping. We might need a conditional import.